src/blockchain/merkle-patricia.js

```diff
@@ -189,46 +206,45 @@ export function merklePatriciaProof(rootHash, proofNode, key, type) {
                 throw new Error('Invalid length of key in tree.');
             }
 
-            if (keySuffix[0] === '0') { // '0' at the beginning means left branch/leaf
-                if (levelData.left === undefined) {
-                    levelData.left = {
-                        hash: branchValueHash,
-                        key: branchKey,
-                        type: branchType,
-                        suffix: keySuffix,
-                        size: fullKey.length
-                    };
-                } else {
-                    throw new Error('Left node is duplicated in tree.');
-                }
-            } else { // '1' at the beginning means right branch/leaf
-                if (levelData.right === undefined) {
-                    levelData.right = {
-                        hash: branchValueHash,
-                        key: branchKey,
-                        type: branchType,
-                        suffix: keySuffix,
-                        size: fullKey.length
-                    };
-                } else {
-                    throw new Error('Right node is duplicated in tree.');
-                }
-            }
+            nodes.push({
+                hash: branchValueHash,
+                key: branchKey,
+                type: branchType,
+                suffix: keySuffix,
+                size: fullKey.length
+            });
+        }
+
+        if (nodes[0].suffix[0] === nodes[1].suffix[0] && keyPrefix.length > 0) {
```

Comment:
Off-topic: are there tests for zero-length `suffix`? As far as I can tell, they aren't deemed invalid by `validateBinaryString` or other checks. With this PR, they will fail ordering, but it could be useful to allocate a couple of dedicated tests in order to prevent regression.